File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <bits/stdc++.h>

using namespace std;

int a, b, c, maxNum;

int main() {
  cin.tie(0) -> ios::sync_with_stdio(0);

  while(true) {
    cin >> a >> b >> c;

    if(a == 0) {
      break;
    }

    maxNum = max(a, max(b, c));

    if(maxNum >= a + b + c - maxNum) {
      cout << "Invalid\n";
    } else if(a == b && b == c && a == c) {
      cout << "Equilateral\n";
    } else if(a != b && b != c && a != c) {
      cout << "Scalene\n";
    } else {
      cout << "Isosceles\n";
    }
  }

  return 0;
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys

input = sys.stdin.readline

ans = []
while True:
    a, b, c = map(int, input().split())
    if a == 0:
        break
    if max(a, b, c) >= a + b + c - max(a, b, c):
        ans.append('Invalid')
        ans.append('\n')
    elif a == b and b == c and c == a:
        ans.append('Equilateral')
        ans.append('\n')
    elif a != b and b != c and c != a:
        ans.append('Scalene')
        ans.append('\n')
    else:
        ans.append('Isosceles')
        ans.append('\n')

print(''.join(ans))